mlflow/utils/environment.py

```diff
@@ -80,5 +84,61 @@ def _get_additional_pip_dep(conda_env):
 
 
 def _log_pip_requirements(conda_env, path, requirements_file="requirements.txt"):
-    pip_deps = _get_additional_pip_dep(conda_env)
+    pip_deps = _get_pip_dep(conda_env)
     _mlflow_additional_pip_env(pip_deps, path=os.path.join(path, requirements_file))
+
+
+def _parse_pip_requirements(pip_requirements):
+    def _is_string(x):
+        return isinstance(x, str)
+
+    def _is_iterable(x):
+        try:
+            iter(x)
+            return True
+        except Exception:
+            return False
+
+    if _is_string(pip_requirements):
+        return [
+            req.requirement for req in parse_requirements(pip_requirements, session=PipSession())
+        ]
+    elif _is_iterable(pip_requirements) and all(map(_is_string, pip_requirements)):
+        try:
+            # Create a temporary requirements file in the current working directory
+            requirements = tempfile.NamedTemporaryFile(
+                mode="w",
+                prefix="mlflow.",
+                suffix=".tmp.requirements.txt",
+                dir=os.getcwd(),
+                delete=False,
+            )
+            requirements.write("\n".join(pip_requirements))
+            requirements.close()
+            return _parse_pip_requirements(requirements.name)
+        finally:
+            # Clena up the temporary requirements file
+            os.remove(requirements.name)
+
+    else:
+        raise TypeError(
+            "Invalid type for `pip_requirements`: {}, must be `str` or iterable of `str`".format(
+                type(pip_requirements)
+            )
+        )
```

Comment:
To clarify, this error can be more specific: `pip_requirements` is either a string path to a pip requirements file on the local filesystem or a list of pip requirement strings.

Reply:
Sound good, I'll update the error message